{Release} Upgrade to Azure CLI 2.69.0 (#30758)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.69.0
++++++

**ACR**

* `az acr manifest list-metadata`: Update help messages for untagged manifests (#30382)

**AKS**

> * `az aks create`: Add parameters `--if-match` and `--if-none-match` to support etag functionality for concurrency (#30309)
> * `az aks update`: Add parameters `--if-match` and `--if-none-match` to support etag functionality for concurrency (#30309)
> * `az aks delete`: Add parameters `--if-match` and `--if-none-match` to support etag functionality for concurrency (#30309)

Reviewer: ```suggestion
* `az aks create/update/delete`: Add parameters `--if-match` and `--if-none-match` to support etag functionality for concurrency (#30309)
```

**App Config**

> * `az appconfig kv import`: Support microsoft feature management schema (#30376)
> * `az appconfig kv export`: Support microsoft feature management schema (#30376)

Reviewer: ```suggestion
* `az appconfig kv import/export`: Support microsoft feature management schema (#30376)
```

* `az appconfig kv export`: Introduce a new environment variable called `AZURE_APPCONFIG_FM_COMPATIBILE` when exporting to a file for backward compatibility for users (#30376)

> * `az appconfig feature show`: Support microsoft feature management schema (#30376)
> * `az appconfig feature list`: Support microsoft feature management schema (#30376)

Reviewer: ```suggestion
* `az appconfig feature show/list`: Support microsoft feature management schema (#30376)
```

* `az appconfig kv restore/show/list`: Update datetime validation to accept timezone offset (#30369)
* `az appconfig revision list`: Update datetime validation to accept timezone offset (#30369)
* `az appconfig export`: Update export help message for environment variable (#30747)

**App Service**

* `az functionapp create`: Refactor EOL implementation and sort based on runtime EOL date (#30636)
* `az functionapp list-flexconsumption-locations`: Check if flex region is enabled for subscription (#30607)
* `az functionapp deployment slot create`: Add `--https-only` parameter for slot creation command (#30628)

> * `az webapp list-runtimes`: Remove the JBoss '_byol' entries from the output for webapps with Linux OS (#30673)

Reviewer 2: ```suggestion
* `az webapp list-runtimes`: Remove the JBoss `_byol` entries from the output for webapps with Linux OS (#30673)
```

**Backup**

* `az backup`: Add support for new AFS Vault Standard Policies (#30531)

**Batch**

* [BREAKING CHANGE] `az batch certificate create/list/show/delete`: Remove deprecated commands (#30501)
* [BREAKING CHANGE] `az batch node reimage/remote-desktop`: Remove deprecated commands (#30501)
* [BREAKING CHANGE] `az batch pool create`: Remove `--application-licenses`, `--certificate-references` , `--os-family` and `--os-version` deprecated parameters (#30501)
* [BREAKING CHANGE] `az batch pool set/reset `: Remove `--certificate-references` deprecated parameter (#30501)

> * `az batch job create`: Add parameter `--job-manager-task-application-package-references` and `--on-all-tasks-complete` (#30501)

Reviewer: ```suggestion
* `az batch job create`: Add parameters `--job-manager-task-application-package-references` and `--on-all-tasks-complete` (#30501)
```

* `az batch job disable`: Add parameter `--json-file` (#30501)

> * `az batch job-schedule create`: Add parameter `--job-manager-task-application-package-references`, `--job-metadata`, and `--job-manager-task-environment-settings` (#30501)

Reviewer: ```suggestion
* `az batch job-schedule create`: Add parameters `--job-manager-task-application-package-references`, `--job-metadata`, and `--job-manager-task-environment-settings` (#30501)
```

> * `az batch job-schedule set/reset`: Add parameter `--job-max-task-retry-count` and `--job-max-wall-clock-time` (#30501)

Reviewer: ```suggestion
* `az batch job-schedule set/reset`: Add parameters `--job-max-task-retry-count` and `--job-max-wall-clock-time` (#30501)
```

* `az batch node reboot`: Add parameter `--json-file` (#30501)
* `az batch node scheduling disable`: Add parameter `--json-file` (#30501)
* `az batch pool autoscale evaluate`: Add parameter `--json-file` (#30501)

> * `az batch pool create`: Add parameter `--start-task-environment-settings` and `--start-task-max-task-retry-count` (#30501)
> * `az batch pool reset`: Add parameter `--start-task-resource-files` and `--target-node-communication-mode` (#30501)

Reviewer: ```suggestion
* `az batch pool create`: Add parameters `--start-task-environment-settings` and `--start-task-max-task-retry-count` (#30501)
* `az batch pool reset`: Add parameters `--start-task-resource-files` and `--target-node-communication-mode` (#30501)
```

**Compute**

* [BREAKING CHANGE] `az sig gallery-application create/update`: Output field `supportedOsType` changed to `supportedOSType` (#30678)
* `az vm list-sizes`: Remove unused parameter `--ids` (#30652)
* `az vmss create/update`: Add new paramter `--zone-balance` to support setting zone balance (#30692)

> * `az vm/vmss create`: Install guest attestation extension when security type set to ConfidentialVM (#30690)

Reviewer 2: ```suggestion
* `az vm/vmss create`: Install guest attestation extension when security type set to `ConfidentialVM` (#30690)
```

* `az vmss scale`: Add new logic to support scaling VMSS in edge zone (#30704)

> * `az vmss create`: Add `--encryption-identity` parameter to use that managed identity for Azure disk encryption (#30657)

Reviewer 2: ```suggestion
* `az vmss create`: Add `--encryption-identity` parameter to use managed identity for Azure disk encryption (#30657)
```

* `az vmss encryption enable`: Add `--encryption-identity` parameter to update or set encryption identity for Azure disk encryption (#30657)

**Container app**

* Fix #28047: `az containerapp compose create`: Upgrade pycomposefile version to split environment variables on the first equal sign instead of every equal sign (#30670)

**DevTest Labs**

> * Add Hibernation Support (#30633)

Reviewer 3: ```suggestion
* `az lab vm`: Add `hibernate` command group (#30633)
```

**Key Vault**

* `az keyvault update/update-hsm`: Fix `--bypass` overriding by default when specifying `--default-action Deny` (#30676)

**Network**

* `az network lb create`: Refine `--frontend-ip-zone` to support multiple zones (#30639)
* `az network private-endpoint-connnection`: Add `Microsoft.HealthDataAiservices/deidservices` for private endpoint connections (#30627)
* `az network routeserver create/update`: Add `--auto-scale-config` (#30702)
* `az network virtual-appliance reimage`: Allow reimage of virtual machines associated with a network virtual appliance (#30680)

**Profile**

* `az login`: Passing the managed identity ID with `--username` is deprecated and will be removed in a future release. Please use `--client-id`, `--object-id` or `--resource-id` instead (#30525)

**RDBMS**

> * `az postgres flexible-server geo-restore`: add `--restore-time` parameter (#30689)

Reviewer: ```suggestion
* `az postgres flexible-server geo-restore`: Add `--restore-time` parameter (#30689)
```

* `az postgres flexible-server fabric-mirroring start/stop/update-databases`: Disable fabric mirroring on HA server (#30688)
* `az postgres flexible-server update`: Fix for scaling up node count on an elastic cluster (#30669)

**Redis**

* `az redis create/update`: Add `--zonal-allocation-policy` to support the way of selecting zones for cache instance (#30705)

**Role**

> * Add `az role definition show` command (#30593)

Reviewer 4: ```suggestion
* `az role definition show`: New command to support showing specific role definition (#30593)
```

**Service Connector**

> * `az webapp connection create redis`: Add `--system-identity` param (#30630)

Reviewer: Keep consistent with the style described above
